File: resolve/horse.py
import sys
import math

inf = 10000000
n = int(raw_input())

power = [];

for i in xrange(n):
    pi = int(raw_input())
    power.append (pi)
power.sort()
i = 0
while i < n-1:
    tmp = power[i+1] - power[i]
    if tmp < inf:
        inf = tmp
    i+=1
    
print inf

File: resolve/horse.c
#include <stdlib.h>
#include <stdio.h>

int main()
{
    int N,i=0,inf=10000000,tmp;
    scanf("%d", &N);
    int power[N];
    while (i< N) {
        scanf("%d", &tmp);
        power[i] = tmp;
        i++;
    }
    
    inline int more (void const *a, void const *b){return ( *(int*)a - *(int*)b );}
    qsort(power, N, sizeof(int), more);

    for (i = 0; i < N-1; i++) {
        tmp = ( power[i+1] - power[i] );
        if (tmp < inf) {
             inf = tmp;
         }
    }
    printf("%d\n",inf);
    return 0;
}